posthog/hogql/printer.py

```diff
@@ -199,95 +301,208 @@ def print_ast(
-            expression, _ = get_property_string_expr(
-                "events",
-                chain[1],
-                escaped_key,
-                "properties",
-            )
-            return HogQLFieldAccess(chain, "event.properties", chain[1], expression)
-        elif chain[0] == "person":
-            if chain[1] in EVENT_PERSON_FIELDS:
-                return HogQLFieldAccess(chain, "person", chain[1], f"person_{chain[1]}")
-            else:
-                raise ValueError(f"Unknown person field '{chain[1]}'")
-    elif len(chain) == 3 and chain[0] == "person" and chain[1] == "properties":
-        key = f"hogql_val_{len(context.values or {})}"
-        context.values[key] = chain[2]
-        escaped_key = f"%({key})s"
-
-        if context.using_person_on_events:
-            expression, _ = get_property_string_expr(
-                "events",
-                chain[2],
-                escaped_key,
-                "person_properties",
-                materialised_table_column="person_properties",
-            )
+    def visit_alias(self, node: ast.Alias):
+        inside = self.visit(node.expr)
+        if isinstance(node.expr, ast.Alias):
+            inside = f"({inside})"
+        return f"{inside} AS {self._print_identifier(node.alias)}"
+
+    def visit_table_symbol(self, symbol: ast.TableSymbol):
+        return self._print_identifier(symbol.table.clickhouse_table())
+
+    def visit_table_alias_symbol(self, symbol: ast.TableAliasSymbol):
+        return self._print_identifier(symbol.name)
+
+    def visit_field_symbol(self, symbol: ast.FieldSymbol):
+        try:
+            last_select = self._last_select()
+            symbol_with_name_in_scope = lookup_field_by_name(last_select.symbol, symbol.name) if last_select else None
+        except ResolverException:
+            symbol_with_name_in_scope = None
+
+        if isinstance(symbol.table, ast.TableSymbol) or isinstance(symbol.table, ast.TableAliasSymbol):
+            resolved_field = symbol.resolve_database_field()
+            if resolved_field is None:
+                raise ValueError(f'Can\'t resolve field "{symbol.name}" on table.')
+            if isinstance(resolved_field, Table):
+                # :KLUDGE: only works for events.person.* printing now
+                if isinstance(symbol.table, ast.TableSymbol):
+                    return self.visit(ast.AsteriskSymbol(table=ast.TableSymbol(table=resolved_field)))
+                else:
+                    return self.visit(
+                        ast.AsteriskSymbol(
+                            table=ast.TableAliasSymbol(
+                                table=ast.TableSymbol(table=resolved_field), name=symbol.table.name
+                            )
+                        )
+                    )
+
+            field_sql = self._print_identifier(resolved_field.name)
+
+            # :KLUDGE: Legacy person properties handling. Assume we're in a context where the tables have been joined,
```

Comment:
I'm not very confident here, but will not block because we'll eventually test all of this with our existing clickhouse insight tests 😬 .

This feels pretty complicated / hard to get right, but don't have a better option on my mind yet, trying to use the existing ColumnOptimizer seems even harder here..

Reply:
Well, we have a test everywhere you can use HogQL property filter (trends, funnels, global filters on all insights, events list, session recordings list, etc) to make sure it works with both `person.properties.email == 'something'` and `properties.something == 'bla'`. Those test are run with materialised properties on and off for those columns, and with person-on-events both on and off. So far they're green. 😅

This all is a big kludge, and I'm now working on a true fix in the context of https://github.com/PostHog/posthog/pull/14286/files